File: samueltm/boredom/math/CustomOperations.java
package samueltm.boredom.math;

public class CustomOperations {

    public static double nthRoot(double base, double root) {
        return Math.pow(base, 1.0 / root);
    }

    public static double geometricMean(double[] numbers) {
        double product = 1;
        for (double n : numbers) {
            product *= n;
        }
        return nthRoot(product, numbers.length);
    }

    public static double arithmeticMean(double[] numbers) {
        double sum = 0;
        for(double n : numbers) {
            sum += n;
        }
        return sum / numbers.length;
    }
}

File: samueltm/boredom/math/CustomOperations.kt
package samueltm.boredom.math

import kotlin.math.pow

@Suppress("MemberVisibilityCanBePrivate")
class CustomOperations {

    companion object {
        fun nthRoot(base: Double, root: Double) = base.pow(1.0 / root)

        fun arithmeticMean(numbers: DoubleArray) = numbers.sum() / numbers.size

        fun geometricMean(numbers: DoubleArray): Double {
            var product = 1.0
            for(n in numbers) {
                product *= n
            }
            return nthRoot(product, numbers.size.toDouble())
        }
    }
}
